src/components/HomeSearchBox/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import { useRouter } from 'next/router'
+import useTranslation from 'next-translate/useTranslation'
+
+const HomeSearchBox = () => {
+  const { t } = useTranslation('home')
+  const router = useRouter()
+  const [searchValue, setSearchValue] = useState<string>('')
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchValue(event.target.value)
+    console.log(searchValue)
+  }
+
+  const handleSearchClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    router.push('products/${searchValue}')
+    setSearchValue('')
+  }
+
+  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && searchValue) {
```

Comment:
Don't think we need this since button will be disabled if the value is empty. Have a try.
```suggestion
    if (event.key === 'Enter') {
```

Reply:
The key is not disabled though.